spec.bs

```diff
@@ -1699,6 +1711,45 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Fence>notifyEvent(|event|)</dfn> method steps are:
+
+  1. If [=this=]'s {{Document}} is not [=Document/fully active=], then return.
+
+  1. Let |navigable| be [=this=]'s [=relevant global object=]'s [=navigable=].
+  
+  1. If any of the following conditions are met, throw a {{SecurityError}} {{DOMException}}:
+
+      * |navigable| is not a [=fenced navigable container/fenced navigable=];
+
+      * |event| is null;
+  
+      * |event|'s {{Event/isTrusted}} is false;
+  
+      * |event|'s [=Event/dispatch flag=] is unset;
+
+      * |event|'s {{Event/type}} is not <code>click</code>
+  
+  1. If [=this=]'s [=relevant global object=] does not have [=transient activation=], then return.
+
+  1. [=Consume user activation=] for [=this=]'s [=relevant global object=].
+
+  1. [=Fire an event=] named <code>[=fencedtreeclick=]</code> at |navigable|'s 
+     [=fenced navigable container=]. Initialize the event's {{Event/bubbles}} and {{Event/cancelable}}
+     attributes to `true`. When running the 
+     <a href="https://dom.spec.whatwg.org/#inner-event-creation-steps">inner event creation steps</a>, 
+     set the <var ignore=''>time</var> to a user-agent-defined value that is consistent across all 
+     invocations of this method.
```

Comment:
Yeah, the idea was to not provide specific timestamps from notifyEvent() to the embedder. Probably not strictly necessary given that the embedder can grab their own close-enough timestamp, but follows the principle of passing as little data to the embedder as possible.

Reply:
Based on conversation in the TPAC breakout, I still think this censored timestamp field is good to keep (even if the parent knows all the information, it's still subject to the time taken between when the fenced frame sends notifyEvent and when the browser actually pops the event listener off of the event loop).

Reply:
So what should the value _actually_ be? Should we just make it always null or something? What does it mean for it to be consistent across all invocations of this method? Certainly a proper "time stamp" can't behave like that, right? (Hence why I ask if we should make it "null")

Reply:
When writing the Chromium implementation, I spent a while thinking about this. Chromium has a TODO to remove [remove](https://source.chromium.org/chromium/chromium/src/+/main:base/time/time.h;drc=4b44bd63028a5aa82976dc4d7ed51cb78cbc6891;l=415) the concept of "null" time. As a result, I used the [Unix epoch](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/dom/events/event.h;drc=a125f7c5687ed22bcd476ce713160096a3d0b037;l=112) instead, since that's a well-known fixed value. I figured given that this timestamp is intentionally a placeholder, its exact value isn't that important.

If we'd like to pick a specific value instead of leaving it up to the user agent, we could just use 0, or the Unix epoch, given that it's defined in the [same spec](https://w3c.github.io/hr-time/#sec-domhighrestimestamp) as DomHighResTimeStamp.

By "consistent across all invocations of this method" I meant: any time a "fencedtreeclick" event is fired, regardless of what the actual current time is, all event objects will have the same timestamp. 

Thinking about it more, I'm leaning toward just using the Unix epoch in this spec directly, to make things less ambiguous.